qt/scientific_interfaces/ISISReflectometry/GUI/Batch/BatchPresenter.cpp

```diff
@@ -139,7 +139,11 @@ void BatchPresenter::notifyAlgorithmComplete(IConfiguredAlgorithm_sptr &algorith
   if (m_savePresenter->shouldAutosave()) {
     auto const workspaces =
         m_jobManager->algorithmOutputWorkspacesToSave(algorithm, m_savePresenter->shouldAutosaveGroupRows());
-    m_savePresenter->saveWorkspaces(workspaces);
+    try {
+      m_savePresenter->saveWorkspaces(workspaces);
+    } catch (...) {
+      // The exception will be logged, we just want to prevent a crash
```

Comment:
It looks like the save presenter handles the exceptions at its level for the most part, so could we be explicit about where these errors are logged that aren't caught in the `saveWorkspaces()` method? If it's not logged there, could we output to the log here?

Reply:
Sorry if you've already realised this, I just thought I'd make sure we're on the same page - the SavePresenter gives us the   `saveSelectedWorkspaces` method, which is the one called when the Save button is used from the Save Ascii tab in the GUI. This method has error handling to pop up a message box when an error occurs. This method internally calls the `saveWorkspaces` method from the SavePresenter, which doesn't have any error handling. The `saveWorkspaces` method is also called directly by the BatchPresenter when we're batch processing with auto-saving switched on. This is the case where Mantid would previously crash if there was any error thrown from the save algorithms - i.e. if we provided an invalid save file prefix (like `?`) on Windows.

I put the error handling at the BatchPresenter level because, among other reasons, it felt like direct callers of the `saveWorkspace` method should decide how to handle exceptions themselves. In this circumstance, the save algorithms are logging errors to the message pane and if we catch the exceptions and log them again then we just get duplicate messages. It might be good to expand this, though, so that we catch and not log anything for the error types we know are thrown by the save algorithms, and use `catch(...)` to log that we couldn't save (but we won't have a stack trace to log). Is that the sort of thing you were thinking here?

Reply:
Alternatively, I'm not a fan of the duplicate logs, but we could just duplicate them to make sure we're getting information about all errors. This potential crash has been in the code for a while and we've never had reports of the users encountering it, so it's unlikely any duplicate logging would create too much of a problem for them.